ui/src/containers/Dashboard/index.js

```diff
@@ -1,29 +1,39 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import * as _ from 'lodash';
 import Scroll from 'react-scroll';
 import { connect } from 'react-redux';
 import withStyles from 'isomorphic-style-loader/lib/withStyles';
 import { actions as routerActions } from 'redux-router5';
-import { withProps, compose, lifecycle } from 'recompose';
+import { withProps, compose, lifecycle, withHandlers, mapProps } from 'recompose';
 import { Map, List, is } from 'immutable';
-import { activeOrgIdSelector } from 'ui/redux/modules/router';
 import Input from 'ui/components/Material/Input';
-import WidgetVisualiseCreator from 'ui/containers/WidgetVisualiseCreator';
-import { withModel } from 'ui/utils/hocs';
+import ConfirmModal from 'ui/components/Modal/ConfirmModal';
+import Spinner from 'ui/components/Spinner';
 import DashboardGrid from 'ui/containers/DashboardGrid';
+import DashboardSharing from 'ui/containers/DashboardSharing';
 import DeleteButton from 'ui/containers/DeleteButton';
 import Owner from 'ui/containers/Owner';
 import PrivacyToggleButton from 'ui/containers/PrivacyToggleButton';
-import DashboardSharing from 'ui/containers/DashboardSharing';
-import Spinner from 'ui/components/Spinner';
+import WidgetVisualiseCreator from 'ui/containers/WidgetVisualiseCreator';
+import { withModel } from 'ui/utils/hocs';
+import { COPY_DASHBOARD } from 'ui/redux/modules/dashboard/copyDashboard';
+import { getVisualisationsFromDashboard } from 'ui/redux/modules/dashboard/selectors';
+import { loggedInUserId } from 'ui/redux/selectors';
+import { activeOrgIdSelector } from 'ui/redux/modules/router';
 import styles from './styles.css';
 
 const schema = 'dashboard';
 
 class Dashboard extends Component {
   static propTypes = {
     model: PropTypes.instanceOf(Map),
-    updateModel: PropTypes.func
+    navigateTo: PropTypes.func,
+    updateModel: PropTypes.func,
+    saveModel: PropTypes.func,
+    setMetadata: PropTypes.func,
+    getMetadata: PropTypes.func,
```

Comment:
I can't see `getMetadata`, `setMetadata`, `saveModel`, and `navigateTo` being used, we might want to remove these?

Reply:
sorry. yes.

Reply:
- `setMetadata` and `getMetadata` is used in toggleSharing (line 137)
- `saveModel` is used createPopulatedWidget (line 117)
- `navigateTo` is passed to DeleteButton (line 187)

Reply:
Okay cool thank you